jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -168,6 +168,26 @@ public Response getSubscription(@PathParam("subscriptionId") final UUID subscrip
         return Response.status(Status.OK).entity(json).build();
     }
 
+    @TimedResource
+    @GET
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Retrieve a subscription by external key", response = SubscriptionJson.class)
+    @ApiResponses(value = {@ApiResponse(code = 404, message = "Subscription not found")})
```

Comment:
For this to work, I believe `EntitlementApiExceptionMapper` needs to be updated to handle the new error code.